commit 68f90970f51015b9410b0786fa4d861089ccd413
Author: Matěj Laitl <matej@laitl.cz>
Date:   Sat Dec 15 23:17:45 2012 +0100

    ServiceBase::setModel(): prevent crash triggered by Last.fm rewrite
    
    BUGFIXES:
     * Prevent crash in ServiceBase::setModel triggered by Last.fm rewrite.
    
    Lukáši, please test the patch so that we have no fear releasing 2.7
    final.
    
    BUG: 311723
    FIXED-IN: 2.7

diff -urN a/amarok-2.6.90/src/services/ServiceBase.cpp b/amarok-2.6.90/src/services/ServiceBase.cpp
--- a/amarok-2.6.90/src/services/ServiceBase.cpp
+++ b/amarok-2.6.90/src/services/ServiceBase.cpp
@@ -173,10 +173,9 @@ ServiceBase::itemActivated ( const QModelIndex & index )
 void
 ServiceBase::setModel( QAbstractItemModel * model )
 {
-    //m_filterModel->setSourceModel( model );
-    //m_contentView->setModel( m_filterModel );
-    m_contentView->setModel( model );
-    m_model  = model;
+    if( m_contentView )
+        m_contentView->setModel( model );
+    m_model = model;
 }
 
 QAbstractItemModel *


